tools/interu/src/config/runner.rs

```diff
@@ -0,0 +1,174 @@
+use std::{fmt::Display, str::FromStr};
+
+use serde::{Deserialize, Serialize};
+use serde_with::{serde_as, DisplayFromStr};
+use snafu::{OptionExt, ResultExt as _, Snafu};
+
+use crate::instances::Instances;
+
+#[serde_as]
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(rename_all = "kebab-case")]
+pub struct Runner {
+    #[serde_as(as = "DisplayFromStr")]
+    pub platform: PlatformPair,
+
+    // TODO (@Techassi): Allow some kind of inheritance here
+    // /// The size of cluster nodes. Can be `small`, `medium`, or `large`.
+    // pub size: Size,
+
+    // /// The size of the disk in GB per cluster node.
+    // pub disk: usize,
+    /// The time-to-live of the cluster.
+    pub ttl: String,
+
+    /// Optionally define node groups.
+    #[serde(default, skip_serializing_if = "Vec::is_empty")]
+    pub node_groups: Vec<NodeGroup>,
+}
+
+#[derive(Debug, Snafu)]
+pub enum ParsePlatformTripleError {
+    #[snafu(display("invalid format, expected pair separated by dashes"))]
+    InvalidFormat,
+
+    #[snafu(display("failed to parse distribution"))]
+    ParseDistribution { source: strum::ParseError },
+}
+
+#[derive(Debug)]
+pub struct PlatformPair {
+    pub distribution: Distribution,
+    // Ideally we want to use SemVer here, but cloud vendors make stupid
+    // decisions and just use major.minor.
+    pub version: String,
+}
+
+impl FromStr for PlatformPair {
+    type Err = ParsePlatformTripleError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let parts: Vec<&str> = s.split('-').collect();
+
+        match parts[..] {
+            [distribution, version] => {
+                let distribution =
+                    Distribution::from_str(distribution).context(ParseDistributionSnafu)?;
+
+                Ok(PlatformPair {
+                    version: version.to_owned(),
+                    distribution,
+                })
+            }
+            _ => InvalidFormatSnafu.fail(),
+        }
```

Comment:
Should we instead use `split_once('-')` and return as a tuple?

```suggestion
        let (distribution, version) = s.split_once('-').context(InvalidFormatSnafu)?;

            let distribution =
                    Distribution::from_str(distribution).context(ParseDistributionSnafu)?;

            Ok(PlatformPair {
                version: version.to_owned(),
                distribution,
            })
        }
```

Reply:
Yes, we should. That piece of code is a left-over from when we expected a platform triple instead of a pair.